### ISSUE TITLE
libpv_leopard.so: wrong ELF class: ELFCLASS64 - Python

### ISSUE THREAD

**Issue by Millertime72** [Picovoice/leopard#215]
I've just reinstalled pvleopard for Python 3.9 on my Raspberry Pi 4. Upon attempting to create a leopard object, I receive an error. It worked fine prior to the reinstallation, but I'm now running v1.2. I tried downgrading to v1.1 and even v1.0, but I just get the same error.
```
import pvleopard

with open('access_key.txt', 'r') as file:
    key = file.read()

leopard = pvleopard.create(access_key=key)
```
Here's the error I receive.
```
Traceback (most recent call last):
  File "/home/pi/Desktop/Project/test.py", line 6, in <module>
    leopard = pvleopard.create(access_key=key)
  File "/home/pi/Desktop/Project/venv/lib/python3.9/site-packages/pvleopard/__init__.py", line 47, in create
    return Leopard(access_key=access_key, library_path=library_path, model_path=model_path)
  File "/home/pi/Desktop/Project/venv/lib/python3.9/site-packages/pvleopard/leopard.py", line 117, in __init__
    library = cdll.LoadLibrary(library_path)
  File "/usr/lib/python3.9/ctypes/__init__.py", line 452, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/pi/Desktop/Project/venv/lib/python3.9/site-packages/pvleopard/lib/raspberry-pi/cortex-a72-aarch64/libpv_leopard.so: wrong ELF class: ELFCLASS64
```
From the error message, I can tell the problem is with the library version, as my platform is 32-bit. Would you mind helping me figure out why pip is installing the wrong version and how I can specify the version that I want installed? Thanks very much.

**Comment by ErisMik** [comment]
Leopard's Python `pip` package ships with the libraries for all platforms, so the issue is a runtime one. To detect the platform the engine is running on we use Python's [`platform`](https://docs.python.org/3/library/platform.html) library. 

Can you please confirm the output of python's `platform.architecture()` and `platform.machine()`?
```
import platform
print(platform.machine())
print(platform.architecture())
```

**Comment by Millertime72** [comment]
I've been doing some digging, and I believe I may have found out what went wrong. I managed to solve my issue by navigating to `lib/python3.9/site-packages/pvleopard/lib/raspberry-pi` and switching the names of the two directories `cortex-a72` and `cortex-a72-aarch64`. The program worked perfectly after that.

I believe the error to be due to the use of the function `platform.machine()`. I saw how it was used in your `_util.py` file to determine the shared object version. The value of `platforme.machine()` stems from the command `uname -m`, which in my case returns `aarch64`, the processor of the Raspberry Pi. However, I am running a 32-bit OS, meaning that the software is of the incorrect version for the shared object, even though the hardware matches. I was able to determine my OS version by running `getconf LONG_BIT`, which returned `32`. It might still be me, but I believe the shared object version should be determined by the software, not the hardware.

Please let me know if I'm misunderstanding, or your general thoughts on the situation. Thanks again.

**Comment by ErisMik** [comment]
I agree, the library needs to match the software not the hardware, however on our test raspberry-pis the value of `uname -m` returns `armv7l` even when running on a 64bit cpu (Raspberry PI 4 w/ 32bit Raspberry Pi OS).

What OS image are you using?

**Comment by Millertime72** [comment]
The default with the Raspberry Pi Imager, "Raspberry Pi OS (32-bit)". It says it's a port of Debian Bullseye.

**Comment by ErisMik** [comment]
Thank you. 

I'm reopening as it appears that recently Raspberry Pi OS changed their KERNEL from 32bit to 64bit, even though the OS remains 64bit which is the reason for the discrepancy you see and between our older test boards. I'll report back and close the issue when a fix is released.

**Comment by ErisMik** [comment]
@Millertime72 A fix has been pushed (`pvleopard 1.2.2`). Can you confirm that this fixes the issue on your device?

**Comment by Millertime72** [comment]
Yes, I can confirm. Thanks very much for your help!
